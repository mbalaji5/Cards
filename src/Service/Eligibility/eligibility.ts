import { IResponse } from "../../DataModel/common";
import {
  IcardEligibility,
  IEligibilityRequest,
} from "../../DataModel/eligibility/eligibility";
import http from "../common";

export const postEligibility = async (reqBody: IEligibilityRequest) => {
  return await http
    .post<IResponse<IcardEligibility>>("/eligibility/check", reqBody)
    .then((data) => data.data)
    .catch((error) => error.response.data);
};
