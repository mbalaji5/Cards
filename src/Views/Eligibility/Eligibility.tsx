import React, { useState } from "react";
import { IResponse } from "../../DataModel/common";
import {
  IcardEligibility,
  IEligibilityRequest,
} from "../../DataModel/eligibility/eligibility";
import View from "../../DesignSystem/View";
import { postEligibility } from "../../Service/Eligibility";
import EligibilityApplication from "./EligibilityApplication";
import EligibilityResults from "./EligibilityResults";

const Eligibility = () => {
  const [eligibilityResult, setEligibilityResult] =
    useState<IResponse<IcardEligibility>>();

  const eligibilityCheckHandler = (reqData: IEligibilityRequest): void => {
    postEligibility(reqData).then((res) => {
      setEligibilityResult(res);
    });
  };

  return (
    <View>
      <EligibilityApplication
        eligibilityCheckHandler={async (reqData) =>
          eligibilityCheckHandler(reqData)
        }
      />
      <EligibilityResults eligibilityResult={eligibilityResult} />
    </View>
  );
};

export default Eligibility;
