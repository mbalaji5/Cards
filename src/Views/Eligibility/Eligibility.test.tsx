import { render } from "@testing-library/react";
import React from "react";
import Eligibility from "./Eligibility";
describe("should validate Eligibility", () => {
  it("Should validate snapshot", () => {
    const { baseElement } = render(<Eligibility />);
    expect(baseElement).toMatchSnapshot();
  });
});
