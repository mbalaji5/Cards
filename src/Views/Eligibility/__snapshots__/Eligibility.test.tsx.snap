// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate Eligibility Should validate snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB iUjGQe"
    >
      <div
        class="sc-gKseQn gTIxDg"
      >
        <h1
          class="sc-jSgvzq dfkKDO"
        >
          Cards
        </h1>
        <form>
          <div
            class="sc-gsTEea eywXce"
          >
            <input
              class="sc-dlfnuX izfRXW"
              data-testid="name"
              id="name"
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-gsTEea eywXce"
          >
            <input
              class="sc-dlfnuX izfRXW"
              data-testid="email"
              id="email"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="sc-gsTEea eywXce"
          >
            <input
              class="sc-dlfnuX izfRXW"
              data-testid="address"
              id="address"
              name="address"
              placeholder="Address"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-hKgJUU eXewpY"
          >
            <button
              class="sc-eCstlR bWdOQs"
              data-testid="submit"
              id="submit"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
        
      </div>
      <div
        class="sc-iBPTik fORSaw"
      >
        
        
      </div>
    </div>
  </div>
</body>
`;
